src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -56,43 +59,67 @@ final class MongoConnection extends ConnectionAdapter {
     @Override
     public void setAutoCommit(boolean autoCommit) throws SQLException {
         checkClosed();
-        throw new NotYetImplementedException(
-                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-30");
+        if (autoCommit != this.autoCommit) {
+            try {
+                if (clientSession.hasActiveTransaction()) {
+                    clientSession.commitTransaction();
+                }
+            } catch (RuntimeException e) {
+                throw new SQLException("Failed to commit transaction", e);
+            }
```

Comment:
One issue with `assertFalse()` is it will throw `Error`, not `Exception`, so it should only used in fatal error case wherein Hibernate will give up totally; but if it is not fatal, Hibernate might handle elegantly if needed for `SQLException` is the contract.

For our current design (no connection pool and no JTA or any transaction customization usage), `assertFalse()` might be fine, but I want to be cautious here.

Reply:
oh, saw we have the precondition of `doCommit()` that `autoCommit` is false, so I think the `assertFalse()` makes sense.

Reply:
well, there could be still edge case that `clientSession.hasActiveTransaction()` is true but `autoCommit` is true. but let us use `assertFalse()` for now.

Reply:
> there could be still edge case that clientSession.hasActiveTransaction() is true but autoCommit is true

You are right, a situation when an auto-committing connection has an active transaction, and `setAutoCommit(false)` is called, seems fine, yet the `assertFalse(autoCommit)` check in `doCommit` fails in such a situation. Let's remove the check.

Reply:
thx. removed.